compiler/modules/nback/number.bal

```diff
@@ -199,36 +218,27 @@ function buildConvertToFloat(llvm:Builder builder, Scaffold scaffold, bir:Conver
         buildConvertIntToFloat(builder, scaffold, val, insn);
         return;
     }
-    else if repr.base != BASE_REPR_TAGGED {
-        return scaffold.unimplementedErr("convert form decimal to float", insn.pos);
+    else if repr.base == BASE_REPR_TAGGED && repr.subtype == t:DECIMAL {
+        buildConvertDecimalToFloat(builder, scaffold, val, insn);
```

Comment:
Why is decimal handled as well as in below runtime function? isn't that redundant?

Reply:
Answering my own question, because result isn't tagged.